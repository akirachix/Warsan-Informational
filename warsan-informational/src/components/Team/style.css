.team-container {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    background-color: #DEF5F1;
    padding: 20px;
  }
  .team-member {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 50px;
    text-align: center;
  }
  .team-member img {
    width: 150px;
    height: 150px;
    border-radius: 50%;
    margin-bottom: 10px;
  }
  .team-member p {
    margin-top: 10px;
    font-size: 1rem;
  }
  @media (max-width: 768px) {
    .team-member img {
      width: 120px;
      height: 120px;
    }
  }
  @media (min-width: 768px) {
    .team-member:nth-child(n+2) {
      flex-basis: 24.33%;
    }
  }
  @media (min-width: 768px) {
    .team-member:nth-child(1) {
      flex-basis: 30.33%;
    }
    /* .team-member:nth-child(4),
    .team-member:nth-child(5) {
      flex-basis: 20%;
    } */
  }









