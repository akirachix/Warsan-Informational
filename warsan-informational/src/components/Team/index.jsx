import React from "react";
import './style.css';

const teamMembers = [
  {
    id: "Aniso",
    name: "Aniso Mohamed Kilas",
    role: "Frontend Developer",
    imageSrc: '/images/aniso.jpg',
  },
  {
    id: "Eunice",
    name: "Eunice Musenyia",
    role: "Full Stack Developer",
    imageSrc: '/images/eunice .jpg',
  },
  {
    id: "Susan",
    name: "Susan Maina",
    role: "Software Developer",
    imageSrc: '/images/susan.jpg',
  },
  {
    id: "Clever",
    name: "Clever Kayitaramirwa",
    role: "Full Stack Developer",
    imageSrc: '/images/clever.jpg',
  },
  {
    id: "Joy",
    name: "Joy Mitingi",
    role: "Mobile Developer",
    imageSrc: '/images/joy.jpg',
  },
];

const Team = () => {
  return (
    <>
      <h1>OUR TEAM</h1><br></br>
      <div className="team-container">
        {teamMembers.map((member) => (
          <div className="team-member" key={member.id} data-testid={member.id}>
            <img src={member.imageSrc} alt={member.id} />
            <p>{member.name}<br></br>{member.role}</p>
          </div>
        ))}
      </div>
    </>
  );
};

export default Team;