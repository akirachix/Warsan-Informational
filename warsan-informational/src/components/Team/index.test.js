import React from 'react';
import { render, screen } from '@testing-library/react';
import Team from './index.jsx';
describe('Team Component', () => {
  it('should render without errors', () => {
    render(<Team />);
    // Check for the presence of the component title
    const teamTitle = screen.getByText('OUR TEAM');
    expect(teamTitle).toBeInTheDocument();
    // Check for each team member's image and name
    const teamMembers = screen.getAllByRole('img');
    expect(teamMembers).toHaveLength(5); // Assuming you have 5 team members
    // Check for specific team member details
    const teamMember1 = screen.getByTestId('Aniso');
    const teamMember2 = screen.getByTestId('Eunice');
    const teamMember3 = screen.getByTestId('Susan');
    const teamMember4 = screen.getByTestId('Clever');
    const teamMember5 = screen.getByTestId('Joy');
    // Check that each team member's image and name are present
    expect(teamMember1).toBeInTheDocument();
    expect(teamMember2).toBeInTheDocument();
    expect(teamMember3).toBeInTheDocument();
    expect(teamMember4).toBeInTheDocument();
    expect(teamMember5).toBeInTheDocument();
    // You can also check for specific team member images by alt text
    const teamMember1Image = screen.getByAltText('Aniso');
    const teamMember2Image = screen.getByAltText('Eunice');
    const teamMember3Image = screen.getByAltText('Susan');
    const teamMember4Image = screen.getByAltText('Clever');
    const teamMember5Image = screen.getByAltText('Joy');
    // Check that each team member's image is present
    expect(teamMember1Image).toBeInTheDocument();
    expect(teamMember2Image).toBeInTheDocument();
    expect(teamMember3Image).toBeInTheDocument();
    expect(teamMember4Image).toBeInTheDocument();
    expect(teamMember5Image).toBeInTheDocument();
  });
});