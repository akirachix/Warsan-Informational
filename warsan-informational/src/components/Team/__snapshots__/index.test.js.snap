// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team Component matches snapshots 1`] = `
<DocumentFragment>
  <h1>
    OUR TEAM
  </h1>
  <br />
  <div
    class="team-container"
  >
    <div
      class="team-member"
      data-testid="Aniso"
    >
      <img
        alt="Aniso"
        src="/images/aniso.jpg"
      />
      <p>
        Aniso Mohamed Kilas
        <br />
        Frontend Developer
      </p>
    </div>
    <div
      class="team-member"
      data-testid="Eunice"
    >
      <img
        alt="Eunice"
        src="/images/eunice .jpg"
      />
      <p>
        Eunice Musenyia
        <br />
        Full Stack Developer
      </p>
    </div>
    <div
      class="team-member"
      data-testid="Susan"
    >
      <img
        alt="Susan"
        src="/images/susan.jpg"
      />
      <p>
        Susan Maina
        <br />
        Software Developer
      </p>
    </div>
    <div
      class="team-member"
      data-testid="Clever"
    >
      <img
        alt="Clever"
        src="/images/clever.jpg"
      />
      <p>
        Clever Kayitaramirwa
        <br />
        Full Stack Developer
      </p>
    </div>
    <div
      class="team-member"
      data-testid="Joy"
    >
      <img
        alt="Joy"
        src="/images/joy.jpg"
      />
      <p>
        Joy Mitingi
        <br />
        Mobile Developer
      </p>
    </div>
  </div>
</DocumentFragment>
`;
